redpanda/templates/configmap.yaml

```
@@ -89,6 +109,9 @@ data:
       - host:
           address: "{{ template "redpanda.fullname" $ }}-{{ . }}.{{ template "redpanda.internal.domain" $ }}"
           port: {{ $values.listeners.rpc.port }}
+{{- end }}
+{{- if not (eq (toYaml .Values.config.node) "null") }}
+  {{ toYaml .Values.config.node | indent 4 }}
```

While testing with `21.11.17` I run into an issue due to this line, where it expects that `.Values.config.node` exists. The issue is that this section shouldn't exist at all in `21.x` configs.

A possible change to solve this would be the following:
```suggestion
{{- if hasKey .Values "config" }}
  {{- if not (eq (toYaml .Values.config.node) "null") }}
    {{ toYaml .Values.config.node | indent 4 }}
  {{- end }}
```

---

@vuldin - I have updated the configmap.yaml to be more tolerant to old versions with newer values.yaml files. New versions will require an explicit config change. Even if the ingest of values into central config is successful the declarative nature of the values file will not necessarily match the config in the central config. Note: upgrading values using the new format with an old values.yaml file was successful.